### PR TITLE
Minor ICN decode speed-up

### PR DESCRIPTION
**Pull request by Districh-ru**
This PR is a part of #7233 and it makes ICN decode a bit faster (about 2-3 %).

### CODE REVIEW

File: src/engine/image_tool.cpp [hunk 1]
```diff
@@ -355,8 +355,15 @@ namespace fheroes2
...
-                memcpy( imageData + posX, data, pixelCount );
-                memset( imageTransform + posX, static_cast<uint8_t>( 0 ), pixelCount );
+                if ( pixelCount > 1 ) {
+                    memcpy( imageData + posX, data, pixelCount );
+                    memset( imageTransform + posX, static_cast<uint8_t>( 0 ), pixelCount );
+                }
+                else {
+                    *( imageData + posX ) = *data;
+                    *( imageTransform + posX ) = static_cast<uint8_t>( 0 );
+                }
```
Comment (ihhub) [inline src/engine/image_tool.cpp:365]: Is it really faster to have a distinguish code for the case of 1 byte? Wouldn't it be slower as usually this command (repeat pixels N times) is usually executed for the number of pixels more than 1.

File: src/engine/image_tool.cpp [hunk 2]
```diff
@@ -371,12 +378,18 @@ namespace fheroes2
...
+                const uint32_t pixelCount = ( countValue != 0 ) ? countValue : *( ++data );
 
                 if ( ( transformValue & 0x40 ) && ( transformType <= 15 ) ) {
-                    memset( imageTransform + posX, transformType, pixelCount );
+                    if ( pixelCount > 1 ) {
+                        memset( imageTransform + posX, transformType, pixelCount );
+                    }
+                    else {
+                        *( imageTransform + posX ) = transformType;
+                    }
```
Comment (ihhub) [inline src/engine/image_tool.cpp:392]: The same comment as above. I am not entirely sure that we would have any benefit in performance by doing this.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @Districh-ru , I left few comments in this pull request. Could you please take a look when you have time?

File: src/engine/image_tool.cpp (hunk 1 above)
Comment (Districh-ru) [reply src/engine/image_tool.cpp]: Hi @ihhub.
In this part only in 15% cases `pixelCount` is equal to 1.
I did test more thorough test by doing many measures of `decodeICNSprite()` computation time doing these actions: start fheroes2, load a save of the "Grand War II" map and cast View All spell (to load all map objects).
And computation speed with these changes and without them is almost the same.
So not to overcomplicate the code I revert these changes.

File: src/engine/image_tool.cpp (hunk 2 above)
Comment (Districh-ru) [reply src/engine/image_tool.cpp]: In this part in 45% cases `pixelCount` is equal to 1.
But here, like in the case above, the computation speed with these changes and without them is also almost the same.
I also revert these changes.

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@Districh-ru , thank you so much for this improvement!

**Pull request merged**